src/NEPTypes.jl

```diff
@@ -179,122 +193,121 @@ julia> compute_Mder(nep,1)-(A0+A1*exp(1))
              end
          end
 
+         if !(eltype(AA) <: SparseMatrixCSC)
+             # Dense
+             this=SPMF_NEP_dense{typeof(AA[1]),Ftype}(n,AA,fii,Schur_fact);
```

Comment:
If we create the following constructor:
```julia
SPMF_NEP_dense{FType}(n, A, fi, Schur_fact) where FType =
    SPMF_NEP_dense{FType, eltype(A)}(n, A, fi, Schur_fact)
```
we should be able to skip the redundant type:
```julia
this = SPMF_NEP_dense{Ftype}(n, AA, fii, Schur_fact)
```
(at the cost of maintaining an extra constructor)

Reply:
Ideally, users would not construct SPMF_NEP_dense themselves, but rely on `SPMF_NEP()`. Your suggestion would encourage direct creation, so I would prefer not to implement this, although the code gets slightly shorter.